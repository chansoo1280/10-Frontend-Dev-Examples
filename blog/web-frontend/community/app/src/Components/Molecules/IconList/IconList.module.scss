@import "@Styles/scss/default.scss";

.icon-list {
    &__count {
        color: $black-45;
    }
}