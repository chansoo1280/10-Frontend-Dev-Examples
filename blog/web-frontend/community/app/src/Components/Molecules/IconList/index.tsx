// #region Global Imports
import React from "react"
import classNames from "classnames"
// #endregion Global Imports

// #region Local Imports
import { defaultProps, Space, Button, Text, Icon } from "@Components"
import styles from "./IconList.module.scss"
import { SpaceProps } from "@Components/Atoms/Space"
import { IconNames } from "@Components/Atoms/Icon/IconNames"
// #endregion Local Imports

export interface IconItem {
    iconName: IconNames
    value: string
}
interface IconListProps extends defaultProps {
    iconList: IconItem[]
    BoxProps?: SpaceProps
}

const IconList = (props: IconListProps): JSX.Element => {
    const { iconList, BoxProps, show, className, ...rest } = props
    const prefixCls = "icon-list"
    const classes = classNames(
        styles[`${prefixCls}`],
        {
            [styles[`${prefixCls}--hide`]]: show === false,
        },
        className,
    )
    return (
        <Space padding="5px 0" gap="16px" className={classes} separator={<div style={{ width: "1px", height: "16px", background: "#0000000F" }}></div>} {...BoxProps} {...rest}>
            {iconList.map(({ iconName, value }) => (
                <Text key={iconName} className={classNames(styles[`${prefixCls}__item`])}>
                    <Icon iconName={iconName} /> {value}
                </Text>
            ))}
        </Space>
    )
}
export default IconList
