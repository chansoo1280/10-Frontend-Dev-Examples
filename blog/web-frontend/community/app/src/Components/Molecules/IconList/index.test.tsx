import { fireEvent, render, screen } from "@testing-library/react"
import IconList from "."
describe("IconList", () => {
    it("renders iconList unchanged", () => {
        const { container } = render(<IconList iconList={[]} />)
        expect(container).toMatchSnapshot()
    })
    it("should render a iconList with the class", () => {
        render(<IconList data-testid="iconList" iconList={[]} />)

        const iconList = screen.getByTestId("iconList")
        expect(iconList).toBeInTheDocument()
        expect(iconList).toHaveClass("space", "icon-list")
    })
})
