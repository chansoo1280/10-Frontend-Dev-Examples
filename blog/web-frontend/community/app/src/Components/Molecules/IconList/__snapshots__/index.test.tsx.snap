// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconList renders iconList unchanged 1`] = `
<div>
  <div
    class="space icon-list"
    style="gap: 16px; padding: 5px 0px;"
  />
</div>
`;
